src/components/notifications/listing/index.js

```diff
@@ -0,0 +1,178 @@
+        ]);
+        setNotificationsMessagesQueryEnabled(true);
+    }, [
+        filter,
+        offset,
+        order,
+        orderBy,
+        rowsPerPage,
+        setNotificationsKey,
+        setNotificationsMessagesQueryEnabled,
+    ]);
+
+    const [markSeenMutation, { isLoading: markSeenLoading }] = useMutation(
+        markSeen,
+        {
+            onSuccess: () => {
+                // TODO update unseen count somehow
+                const newPage = {
+                    ...notifications,
+                    messages: notifications.messages.map((n) =>
+                        selected.includes(n.message.id)
+                            ? { ...n, seen: true }
+                            : n
+                    ),
+                };
+
+                setMarkAsSeenEnabled(false);
+                setNotifications(newPage);
+                queryCache.setQueryData(notificationsKey, newPage);
+            },
+            onError: (error) => {
+                showErrorAnnouncer(
+                    t("errorMarkAsSeen", {
+                        count: selected.length,
+                    }),
+                    error
+                );
+            },
+        }
+    );
+
+    const [
+        deleteNotificationsMutation,
+        { isLoading: deleteLoading },
+    ] = useMutation(deleteNotifications, {
+        onSuccess: () =>
+            queryCache.invalidateQueries(NOTIFICATIONS_MESSAGES_QUERY_KEY),
+        onError: (error) => {
+            showErrorAnnouncer(
+                t("errorNotificationDelete", {
+                    count: selected.length,
+                }),
+                error
+            );
+        },
+    });
+
+    const handleMarkSeenClick = () => {
+        markSeenMutation(selected);
+    };
+
+    const handleDeleteClick = () => {
+        deleteNotificationsMutation(selected);
+    };
+
+    const handleFilterChange = (event) => {
+        setFilter(event.target.value);
+    };
+
+    const onSelectionChanged = (newSelected) => {
+        setSelected(newSelected);
+        setMarkAsSeenEnabled(
+            !notifications.messages.find(
+                (n) => newSelected.includes(n.message.id) && n.seen
+            )
+        );
+    };
+
+    const baseId = baseDebugId + ids.NOTIFICATION_VIEW;
+
+    const hasSelection = selected.length > 0;
+
+    return (
+        <>
+            <NotificationToolbar
+                baseDebugId={baseId}
+                filter={filter}
+                onFilterChange={handleFilterChange}
+                markAsSeenEnabled={hasSelection && markAsSeenEnabled}
+                deleteEnabled={hasSelection}
+                onMarkSeenClicked={handleMarkSeenClick}
+                onDeleteClicked={handleDeleteClick}
+            />
+            <TableView
+                baseId={baseId}
+                data={notifications?.messages}
+                error={error}
+                loading={isFetching || markSeenLoading || deleteLoading}
```

Comment:
Minor suggestion - Since `mark as seen` is not really changing other that row highlighting from users perspective, we don't need a loading mask or skeleton. This could be silent background operation.

Reply:
Good point.
I guess if the service returns an error, I can reset the highlight on those rows (assuming the user hasn't navigated away already) 🤔

Reply:
And I do see logic to display an error message 👍

Reply:
The `Mark as Seen` action will no longer mask the listing with a loading skeleton, and will instantly mark the selected rows as seen. If an error is returned from the service, then the listing will now reload.

I also added logic so the `Mark as Seen` enabled flag will now update whenever the listing or the selection changes. That way if an error occurs and the user hasn't changed their selection or the page they are viewing, then the `Mark as Seen` button should be re-enabled so they can try again.